src/main/views/macros/sjp-common.njk

```diff
@@ -117,23 +117,32 @@
 
             filters.forEach(function(filter, index) {
                 let checkboxId = filter.firstElementChild.id;
+                let searchFilters = document.getElementById('search-filters');
                 if (isPostcodeFilter(filter)) {
+                    let currentQueryArray = window.location.search.split('&');
+                    let currentFilterValues = getCurrentFilterValues(currentQueryArray);
                     if (searchFilters.value.length > 0) {
                         postalArea = extractPostalArea(filter);
                         if ((districtMatch(postalArea, searchFilters, filter)) || (areaMatch(postalArea, searchFilters))) {
```

Comment:
I wonder if you can replace this entire if condition by just checking "filterIncludesSearchTerm(filter, searchFilters)". If so we could remove districtMatch() and areaMatch(). There is no district anymore if we are not checking for postcode area